pallets/pips/src/lib.rs

```diff
@@ -1244,6 +1274,32 @@ impl<T: Trait> Module<T> {
             .collect::<Vec<_>>()
     }
 
+    /// Retrieve historical voting of `who` account.
+    pub fn voting_history_by_address(who: T::AccountId) -> HistoricalVoting<Vote<BalanceOf<T>>> {
+        <ProposalMetadata<T>>::iter()
+            .map(|meta| VoteByPip {
+                pip: meta.id,
+                vote: Self::proposal_vote(meta.id, &who),
+            })
+            .collect::<Vec<_>>()
```

Comment:
I don't think iterating on the backend is beneficial when the same can be done on the client. Such unbounded iterations can become dos vectors.